en/developer/reference/articles/02-web-experience-management/02-page-fragments/02-fragment-configuration-types.markdown

```diff
@@ -89,47 +89,101 @@ If you were to choose the color white, the `h3` tag heading would have the class
-                    "type": "itemSelector",
-                    "typeOptions": {
-                        "className": "com.liferay.journal.model.JournalArticle",
-                        "enableSelectTemplate": true
-                    }
-                }
-            ]
-        }
-    ]
+	"fieldSets": [{
+		"fields": [{
+			"label": "select-content",
+			"name": "itemSelector1",
+			"type": "itemSelector",
+			"typeOptions": {
+				"enableSelectTemplate": true
+			}
+		}]
+	}]
 }
 ```
 
-The only *required* properties for this case are
+You can provide a more advanced configuration that lets authors select only a specific type of content. The configuration below specifies that only web content articles can be selected. The optional `itemSubtype` property specfies that the selected web content article must use the structure `article-structure-key-15` to be selected:
 
 ```json
-"name": "itemSelector1",
-"type": "itemSelector",
-"typeOptions": {
-    "className": "com.liferay.journal.model.JournalArticle"
+{
+	"fieldSets": [{
+		"fields": [{
+			"label": "select-content",
+			"name": "itemSelector1",
+			"type": "itemSelector",
+			"typeOptions": {
+        "itemType" : "com.liferay.journal.model.JournalArticle",
+        "itemSubtype": "article-structure-key-15"
+			}
+		}]
+	}]
 }
 ```
 
-This example provides options to select existing web content articles to display
-in your Fragment.
+This example specifies that only a document with the `img` or `jpg` MIME type that uses the structure `metadataset-structure-key-2` can be selected:
+
+```json
+{
+	"fieldSets": [{
+		"fields": [{
+			"label": "select-content",
+			"name": "itemSelector1",
+			"type": "itemSelector",
+			"typeOptions": {
+        "itemType" : "com.liferay.portal.kernel.repository.model.FileEntry",
+        "itemSubtype": "metadataset-structure-key-2",
+        "mimeTypes": ["img/jpg"]
+			}
+		}]
+	}]
+}
+```
+
+This example specifies that only blog entries can be selected:
+
+```json
+{
+	"fieldSets": [{
+		"fields": [{
+			"label": "select-content",
+			"name": "itemSelector1",
+			"type": "itemSelector",
+			"typeOptions": {
+        "itemType" : "com.liferay.blogs.model.BlogsEntry",
+			}
+		}]
+	}]
+}
+```
+
+You can then render the content in your fragment with this HTML snippet for the web content article:
+
+```markup
+<div class="fragment_name">
+  [#if configuration.itemSelector1.content??]
+       ${configuration.itemSelector1.content}
+  [/#if]
+</div>
+```
+
+You can also access the Java object in your fragment, if you need access to specific portions of the content, under the key `[name-of-field]Object` (`itemSelector1Object` 
+in the example below). This example renders the title, description, and body of the web content article:
+
+```markup
+<div class="fragment_name">
+  [#if configuration.itemSelector1.content??]
+    ${itemSelector1Object.getTitle()}
+    ${itemSelector1Object.getDescription()}
+    ${itemSelector1Object.getContent()?html}
```

Comment:
I would not append ?html here, this would shows a horrible xml, not very appropiate for a sample